ourhexenv.py

```diff
@@ -208,14 +215,41 @@ def step(self, action):
                 self.rewards = {agent: move_r for agent in self.agents}
                 self.terminations = {agent: False for agent in self.agents}
 
-        if self.agent_selection == "player_2":
-            # Player 2 has made their first move, make pie rule unusable
-            self.is_pie_rule_usable = False
-
+        # Update cumulative rewards
         for agent in self.agents:
             self._cumulative_rewards[agent] += self.rewards[agent]
 
-        #Advance to the next agent if the game is not over
+        # Prepare observation space
+        if self.pie_rule_used:
+            # Flipping the observation space if the pie rule was used
```

Comment:
Again, I don't think we "flip" the board. The pie rule just swaps the stone marked as "1" for a "2".